example/main.go

```diff
@@ -54,18 +54,39 @@ func autoCompleteCallback(t *terminal.Terminal, line string, pos int, key rune)
 	return
 }
 
+func AddOrReplaceHistory(t *terminal.Terminal, replace bool, l string) {
+	// if in default auto mode, we don't manage history
+	// we also don't add empty commands. (at start of program)
+	if t.AutoHistory() || l == "" {
```

Comment:
You could also add something to do not log command starting with space, as it's a common pattern to ignore thing in history

Reply:
never heard of leading space preventing history before:
```
$    echo "this had space"
$  history | tail
 568     echo "this had space"
 569  history | tail
```

also I do want spaces in history for grol indentation

Reply:
https://unix.stackexchange.com/questions/115917/why-is-bash-not-storing-commands-that-start-with-spaces

The line is removed when stored, not in local history